libs/ts-rest/core/src/lib/client.ts

```diff
@@ -200,15 +204,30 @@ export const tsRestFetchApi: ApiFetcher = async ({
   });
   const contentType = result.headers.get('content-type');
 
-  if (contentType?.includes('application/json')) {
-    return { status: result.status, body: await result.json() };
-  }
+  const responseHeaders: Record<string, string> = {};
+  result.headers.forEach((value, key) => {
+    responseHeaders[key] = value;
+  });
```

Comment:
I think it might be slightly more DX-friendly is we just return the entire `result.headers` objects directly from fetch, and type the headers property as `Headers` rather than `Record<string, string>`.

The problem is that an object is case sensitive so something like this `{ 'content-type': 'text/plain' }['Content-Type'] } would return `undefined`, but with the Headers interface, you'd use `headers.get('Content-Type')` and it'd be case insensitive. To make sure this isn't a problem, we'd need to normalize all headers as lowercase and make sure users only try to get lowercase headers. Even more of problem, to make sure custom clients normalize the headers themselves.

The Headers interface is part of the Web API spec so it's available on pretty much all browsers. For custom clients like axios it can be easily initialized with `new Headers(response.headers.toJSON())`

Reply:
Yeah, very good point! Will get that sorted :)